app/Models/HasDOI.php

```diff
@@ -145,17 +153,19 @@ public function generateDOI($doiService)
                     'creators' => count($creators) > 0 ? $creators : $contributors,
                     'titles' => [
                         [
-                            'title' => $this->name,
+                            'title' => $title,
```

Comment:
$title is not in the scope, this will always be null unless you pre-initialise the variable in the outer block. Please check